src/main/java/org/broadinstitute/consent/http/mail/message/DarExpirationReminderMessage.java

```diff
@@ -0,0 +1,32 @@
+package org.broadinstitute.consent.http.mail.message;
+
+import java.util.Map;
+import org.broadinstitute.consent.http.enumeration.EmailType;
+import org.broadinstitute.consent.http.models.User;
+
+public class DarExpirationReminderMessage extends MailMessage {
+
+  private final String darCode;
+
+  public DarExpirationReminderMessage(User toUser, String darCode) {
+    super(toUser, EmailType.DAR_EXPIRATION_REMINDER);
+    this.darCode = darCode;
+  }
+
+  @Override
+  public String createSubject() {
+    return "Data Access Request Expiration Reminder for %s".formatted(darCode);
+  }
+
+  @Override
+  public Object createModel(String serverUrl) {
+    return Map.of("userName", toUser.getDisplayName(),
+        "darCode", darCode,
+        "serverUrl", serverUrl);
+  }
+
+  @Override
+  public String getEntityReferenceId() {
+    return darCode;
```

Comment:
We can't use this as the key because darCodes get reused in progress reports.  I'm modifying @pshapiro4broad 's reference implementation to include a referenceId field so we can return the needed value in getEntityReferenceId().  I'm setting the referenceId value in the constructor.

Reply:
Nice catch, yes, we should try to be as granular as possible in referencing entity ids in email entries.